bliss/unconstrained_dists.py

```diff
@@ -40,26 +40,43 @@ def __init__(self, mu, sigma):
         super().__init__(validate_args=False)
 
         multiple_normals = Normal(mu, sigma)
-        prob_in_unit_box_hw = multiple_normals.cdf(torch.ones_like(mu))
-        prob_in_unit_box_hw -= multiple_normals.cdf(torch.zeros_like(mu))
-        self.log_prob_in_unit_box = prob_in_unit_box_hw.log().sum(dim=-1)
-
         self.base_dist = Independent(multiple_normals, 1)
 
+        prob_in_unit_box_hw = multiple_normals.cdf(self.b) - multiple_normals.cdf(self.a)
+        self.log_prob_in_unit_box = prob_in_unit_box_hw.log().sum(dim=-1)
+
     def __repr__(self):
         return f"{self.__class__.__name__}({self.base_dist.base_dist})"
 
-    def sample(self, **args):
-        # some ideas for how to sample it here, if we need to:
-        # https://cran.r-project.org/web/packages/truncnorm/
-        raise NotImplementedError("sampling a truncated normal isn't straightforward")
+    def sample(self, sample_shape=(1,), **args):
+        """Generate sample using rejection sampling."""
+        q = Independent(Normal(self.base_dist.mean, self.base_dist.stddev), 1)
+        samples = q.sample(sample_shape)
+        valid = (samples.min(dim=-1)[0] >= 0) & (samples.max(dim=-1)[0] < 1)
+        while not valid.all():
+            new_samples = q.sample(sample_shape)
+            samples[~valid] = new_samples[~valid]
```

Comment:
Maybe add a comment explaining this is inefficient, but good enough for now, and that we could do better by filtering the invalid samples from `new_samples` *before* replacing old invalid one with new ones.

Reply:
Done.